clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -697,21 +695,20 @@ public Node leastLoadedNode(long now) {
         }
     }
 
-    public static AbstractResponse parseResponse(ByteBuffer responseBuffer, RequestHeader requestHeader) {
-        Struct responseStruct = parseStructMaybeUpdateThrottleTimeMetrics(responseBuffer, requestHeader, null, 0);
-        return AbstractResponse.parseResponse(requestHeader.apiKey(), responseStruct,
-                requestHeader.apiVersion());
+    public static AbstractResponse parseResponse(RequestHeader requestHeader, ByteBuffer responseBuffer) {
+        return parseResponse(requestHeader, responseBuffer, null, 0);
     }
 
-    private static Struct parseStructMaybeUpdateThrottleTimeMetrics(ByteBuffer responseBuffer, RequestHeader requestHeader,
-                                                                    Sensor throttleTimeSensor, long now) {
+    private static AbstractResponse parseResponse(RequestHeader requestHeader, ByteBuffer responseBuffer,
+                                                  Sensor throttleTimeSensor, long now) {
         ResponseHeader responseHeader = ResponseHeader.parse(responseBuffer, requestHeader.headerVersion());
         // Always expect the response version id to be the same as the request version id
-        Struct responseBody = requestHeader.apiKey().parseResponse(requestHeader.apiVersion(), responseBuffer);
+        AbstractResponse response = AbstractResponse.parseResponse(requestHeader.apiKey(), responseBuffer,
+            requestHeader.apiVersion());
         correlate(requestHeader, responseHeader);
-        if (throttleTimeSensor != null && responseBody.hasField(CommonFields.THROTTLE_TIME_MS))
-            throttleTimeSensor.record(responseBody.get(CommonFields.THROTTLE_TIME_MS), now);
-        return responseBody;
+        if (throttleTimeSensor != null && response.throttleTimeMs() != AbstractResponse.DEFAULT_THROTTLE_TIME)
```

Comment:
Need to verify that this condition is equivalent to the previous behavior of checking if the field exists.